### DISCUSSION TITLE
Return text-to-speech for use in games

### DISCUSSION THREAD

**Discussion by kveberg** [godotengine/godot-proposals#10232] [Engine Core]
Calls to TTS return the audio so it can be passed to an audio bus in Godot so it can be further processed for better integration in the overall audio and feel of games. Your proverbial "AI" voice in game, for instance, can be achieved with a little processing (e.g. reverb) to greatly enhance immersion.

**Comment by Calinou** [comment]
[Text-to-speech is already implemented in 4.0 and later](https://github.com/godotengine/godot/pull/56192), but it uses the OS' accessibility APIs. These are not designed to provide a consistent result across platforms, as different platforms have different speech backends (and different voices installed).

The OS itself is playing back the text-to-speech audio, so there is no way Godot could possibly retrieve the output of TTS. This allows TTS to work even if Godot can't play audio for some reason (e.g. if it's using the Dummy audio driver). It also allows adjusting the TTS volume independently from the game volume in the system audio mixer.
